### PR TITLE
Add shared changes from corert

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/mscorlib/shared/System/Collections/Generic/ICollection.cs [hunk 1]
```diff
@@ -26,9 +14,8 @@ namespace System.Collections.Generic
     // Note that T[] : IList<T>, and we want to ensure that if you use
```
Comment (jkotas) [inline src/mscorlib/shared/System/Collections/Generic/ICollection.cs]: This comment is not relevant for CoreCLR or CoreRT. Could you please delete it?

File: src/mscorlib/shared/System/Collections/Generic/IEnumerable.cs [hunk 2]
```diff
@@ -27,9 +15,8 @@ namespace System.Collections.Generic
     // Note that T[] : IList<T>, and we want to ensure that if you use
     // IList<YourValueType>, we ensure a YourValueType[] can be used
```
Comment (jkotas) [inline src/mscorlib/shared/System/Collections/Generic/IEnumerable.cs]: Irrelevant comment, could you please delete it?

File: src/mscorlib/shared/System/Collections/Generic/IList.cs [hunk 3]
```diff
@@ -28,9 +16,8 @@ namespace System.Collections.Generic
     // Note that T[] : IList<T>, and we want to ensure that if you use
     // IList<YourValueType>, we ensure a YourValueType[] can be used 
     // without jitting.  Hence the TypeDependencyAttribute on SZArrayHelper.
```
Comment (jkotas) [inline src/mscorlib/shared/System/Collections/Generic/IList.cs]: Same here.

File: src/mscorlib/shared/System/Lazy.cs [hunk 4]
```diff
@@ -183,6 +183,7 @@ internal static LazyThreadSafetyMode GetModeFromIsThreadSafe(bool isThreadSafe)
     /// </para>
     /// </remarks>
     [Serializable]
+    [ComVisible(false)]
```
Comment (jkotas) [inline src/mscorlib/shared/System/Lazy.cs]: Undo this. It is a recent change made by somebody else - this should be deleted in the CoreRT copy instead.

(It is an example of change that the mirror would take care of propagating if we had one.)

Comment (alexperovich) [inline src/mscorlib/shared/System/Lazy.cs]: Yep, I am in the process of getting that up and running.

**Pull request merged**